sympy/core/function.py

```diff
@@ -707,8 +707,7 @@ def fdiff(self, argindex=1):
             else:
                 return Derivative(self, self.args[argindex - 1], evaluate=False)
         # See issue 4624 and issue 4719 and issue 5600
-        arg_dummy = Dummy('xi_%i' % argindex)
-        arg_dummy.dummy_index = hash(self.args[argindex - 1])
+        arg_dummy = Dummy('xi_%i' % argindex, dummy_index=hash(self.args[argindex - 1]))
```

Comment:
@smichr This kind of thing should be allowed: here its so the *same* Dummy `xi1` gets used in multiple places in the chain rule:
````
>>> f(g(x)).diff(x)
Derivative(g(x), x)*Subs(Derivative(f(_xi_1), _xi_1), (_xi_1,), (g(x),))
>>> pprint(_)
d        ⎛ d        ⎞│       
──(g(x))⋅⎜───(f(ξ₁))⎟│       
dx       ⎝dξ₁       ⎠│ξ₁=g(x)
````
So we may have to allow pretty much arbritary things for `dummy_index` (or otherwise undo this reasonable-looking change)